python/ray/experimental/channel/gpu_communicator.py

```diff
@@ -12,6 +22,25 @@
 # (shape: Tuple[int], dtype: torch.dtype) -> torch.Tensor.
 TorchTensorAllocator = Callable[[Tuple[int], "torch.dtype"], "torch.Tensor"]
 
+T = TypeVar("T")
+
+
+@DeveloperAPI
+class GPUFuture(ABC, Generic[T]):
+    """
+    Future for a GPU operation.
+    """
+
+    @abstractmethod
+    def wait(self, waiter: Optional[Any]) -> T:
```

Comment:
Wouldn't this also need to take in the current stream?

In this case, might be better to take the current stream from `ChannelContext` instead of as an arg.

Reply:
My understanding is current stream may change at different point in time? How could we keep it up to date in `ChannelContext` or when it is passed in?

Reply:
Fixed.